Core/src/Material/SurfaceMaterialMapper.cpp

```diff
@@ -425,9 +425,14 @@ void Acts::SurfaceMaterialMapper::mapInteraction(
   for (auto tmapBin : touchedMapBins) {
     std::vector<std::array<std::size_t, 3>> trackBins = {tmapBin.second};
     if (m_cfg.computeVariance) {
-      tmapBin.first->trackVariance(
-          trackBins, touchedMaterialBin[tmapBin.first]->materialSlab(
-                         trackBins[0][0], trackBins[0][1]));
+      // This only makes sense for the binned material
+      auto binnedMaterial = dynamic_cast<const BinnedSurfaceMaterial*>(
+          touchedMaterialBin[tmapBin.first].get());
```

Comment:
would it make sense to distinguish between non-`BinnedSurfaceMaterial` and `nullptr`?

Reply:
The following PRs (from the MM) series, will actually clean this up.

I.e. you the `SurfaceMaterialMapper` will turn into `BinnedSurfaceMaterialAcummulator` - so the type will be clear here.